🐛 BUG: ✘ [ERROR] This command cannot be run in a non-interactive context

### Which Cloudflare product(s) does this pertain to?

C3

### What version(s) of the tool(s) are you using?

^3.0.0

### What version of Node are you using?

v18.18.2

### What operating system are you using?

Mac

### Describe the Bug

I tried following the Getting Started Tutorial: https://developers.cloudflare.com/workers/get-started/guide/

When i run `npm create cloudflare@latest` I choose a directory, choose "Hello World" Worker, and choose Typescript "yes". 
The script fails with 

✘ [ERROR] This command cannot be run in a non-interactive context
 ⛅️ wrangler 3.14.0
-------------------
Total Upload: 0.19 KiB / gzip: 0.16 KiB
▲ [WARNING] You need to register a workers.dev subdomain before publishing to workers.dev


? Would you like to register a workers.dev subdomain now?
🤖 Using default value in non-interactive context: yes

✘ [ERROR] This command cannot be run in a non-interactive context


If you think this is a bug then please create an issue at https://github.com/cloudflare/workers-sdk/issues/new/choose

### Please provide a link to a minimal reproduction

_No response_

### Please provide any relevant error logs

_No response_

--- a/.changeset/smart-drinks-joke.md
+++ b/.changeset/smart-drinks-joke.md
@@ -1,6 +1,5 @@
 ---
 "create-cloudflare": patch
-"wrangler": patch
 ---
 
 Make sure `wrangler deploy` and `wrangler pages deploy` are run in an interactive context in C3, so that users are able to respond to interactive prompts

--- a/.changeset/smart-drinks-joke.md
+++ b/.changeset/smart-drinks-joke.md
@@ -1,0 +1,6 @@
+---
+"create-cloudflare": patch
+"wrangler": patch
+---
+
+Make sure `wrangler deploy` and `wrangler pages deploy` are run in an interactive context in C3, so that users are able to respond to interactive prompts

--- a/packages/create-cloudflare/src/__tests__/deploy.test.ts
+++ b/packages/create-cloudflare/src/__tests__/deploy.test.ts
@@ -130,8 +130,10 @@ describe("deploy helpers", async () => {
 			ctx.template.platform = "pages";
 			ctx.commitMessage = commitMsg;
 			mockInsideGitRepo(false);
-			vi.mocked(runCommand).mockResolvedValueOnce(deployedUrl);
-
+			vi.mocked(runCommand).mockResolvedValueOnce("");
+			vi.mocked(readFile).mockImplementationOnce(
+				() => `{"type":"deploy", "targets":["${deployedUrl}"]}`,
+			);
 			await runDeploy(ctx);
 			expect(runCommand).toHaveBeenCalledWith(
 				["npm", "run", "deploy", "--", "--commit-message", `"${commitMsg}"`],

--- a/packages/create-cloudflare/src/deploy.ts
+++ b/packages/create-cloudflare/src/deploy.ts
@@ -1,7 +1,7 @@
 import { mkdtemp } from "node:fs/promises";
 import { tmpdir } from "node:os";
 import { join } from "node:path";
-import { crash, startSection, updateStatus } from "@cloudflare/cli";
+import { startSection, updateStatus } from "@cloudflare/cli";
 import { blue, brandColor, dim } from "@cloudflare/cli/colors";
 import TOML from "@iarna/toml";
 import { processArgument } from "helpers/args";
@@ -136,10 +136,10 @@ export const runDeploy = async (ctx: C3Context) => {
 		if (deployedUrlMatch) {
 			ctx.deployment.url = deployedUrlMatch[0];
 		} else {
-			crash("Failed to find deployment url.");
+			throw new Error("Failed to find deployment url.");
 		}
 	} catch {
-		crash("Failed to find deployment url.");
+		throw new Error("Failed to find deployment url.");
 	}
 
 	// if a pages url (<sha1>.<project>.pages.dev), remove the sha1

--- a/packages/create-cloudflare/src/deploy.ts
+++ b/packages/create-cloudflare/src/deploy.ts
@@ -108,7 +108,7 @@ export const runDeploy = async (ctx: C3Context) => {
 		"output.json",
 	);
 
-	const result = await runCommand(deployCmd, {
+	await runCommand(deployCmd, {
 		cwd: ctx.project.path,
 		env: {
 			CLOUDFLARE_ACCOUNT_ID: ctx.account.id,

--- a/packages/create-cloudflare/src/deploy.ts
+++ b/packages/create-cloudflare/src/deploy.ts
@@ -1,9 +1,13 @@
-import { startSection, updateStatus } from "@cloudflare/cli";
+import { mkdtemp } from "node:fs/promises";
+import { tmpdir } from "node:os";
+import { join } from "node:path";
+import { crash, startSection, updateStatus } from "@cloudflare/cli";
 import { blue, brandColor, dim } from "@cloudflare/cli/colors";
 import TOML from "@iarna/toml";
 import { processArgument } from "helpers/args";
 import { C3_DEFAULTS, openInBrowser } from "helpers/cli";
 import { quoteShellArgs, runCommand } from "helpers/command";
+import { readFile } from "helpers/files";
 import { detectPackageManager } from "helpers/packageManagers";
 import { poll } from "helpers/poll";
 import { isInsideGitRepo } from "./git";
@@ -99,12 +103,17 @@ export const runDeploy = async (ctx: C3Context) => {
 			: []),
 	];
 
+	const outputFile = join(
+		await mkdtemp(join(tmpdir(), "c3-wrangler-deploy-")),
+		"output.json",
+	);
+
 	const result = await runCommand(deployCmd, {
-		silent: true,
 		cwd: ctx.project.path,
 		env: {
 			CLOUDFLARE_ACCOUNT_ID: ctx.account.id,
 			NODE_ENV: "production",
+			WRANGLER_OUTPUT_FILE_PATH: outputFile,
 		},
 		startText: "Deploying your application",
 		doneText: `${brandColor("deployed")} ${dim(
@@ -112,12 +121,25 @@ export const runDeploy = async (ctx: C3Context) => {
 		)}`,
 	});
 
-	const deployedUrlRegex = /https:\/\/.+\.(pages|workers)\.dev/;
-	const deployedUrlMatch = result.match(deployedUrlRegex);
-	if (deployedUrlMatch) {
-		ctx.deployment.url = deployedUrlMatch[0];
-	} else {
-		throw new Error("Failed to find deployment url.");
+	try {
+		const contents = readFile(outputFile);
+
+		const entries = contents
+			.split("\n")
+			.filter(Boolean)
+			.map((entry) => JSON.parse(entry));
+		const url: string | undefined =
+			entries.find((entry) => entry.type === "deploy")?.targets?.[0] ??
+			entries.find((entry) => entry.type === "pages-deploy")?.url;
+		const deployedUrlRegex = /https:\/\/.+\.(pages|workers)\.dev/;
+		const deployedUrlMatch = url?.match(deployedUrlRegex);
+		if (deployedUrlMatch) {
+			ctx.deployment.url = deployedUrlMatch[0];
+		} else {
+			crash("Failed to find deployment url.");
+		}
+	} catch {
+		crash("Failed to find deployment url.");
 	}
 
 	// if a pages url (<sha1>.<project>.pages.dev), remove the sha1

